internal/engine/upper.go

```diff
@@ -422,13 +423,38 @@ func handleStartProfilingAction(state *store.EngineState) {
 	state.IsProfiling = true
 }
 
+// The Flag Config is sometimes written to by Tilt itself, and we don't want that to trigger
+// a Tiltfile execution. If the file's timestamp predates the time that Tilt wrote to the file,
+// assume that the notification is just for something Tilt already knows about and ignore it.
+func ignoreFlagConfigUpdate(state *store.EngineState) (ignored bool, err error) {
+	st, err := os.Stat(state.FlagsState.ConfigPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return false, nil
+		} else {
+			return true, err
+		}
+	}
+	return st.ModTime().Before(state.FlagsState.LastArgsWrite), nil
+}
+
 func handleFSEvent(
 	ctx context.Context,
 	state *store.EngineState,
 	event targetFilesChangedAction) {
 
 	if event.targetID.Type == model.TargetTypeConfigs {
 		for _, f := range event.files {
+			if f == state.FlagsState.ConfigPath {
+				ignored, err := ignoreFlagConfigUpdate(state)
```

Comment:
we try pretty hard to make the reducers pure. they shouldn't read file system state. can you move this check into the subscriber?

Reply:
Oh, I was unaware of this. I've been calling `time.Now` willy-nilly. Good to know.